lib/src/model/puzzle/puzzle_controller.dart

```diff
@@ -281,12 +281,12 @@ class PuzzleController extends _$PuzzleController with EngineEvaluationMixin {
 
   void _saveCurrentStreakLocally() {
     if (state.streak != null) {
-      ref.read(streakStorageProvider(initialContext.userId)).saveActiveStreak(state.streak!);
+      ref.read(streakStorageProvider(initialContext.userId)).saveActiveStreak(ref, state.streak!);
```

Comment:
That is weird to pass the `ref` like that, especially since the StreakStorage is a provider.

Idiomatic way with riverpod is to pass the `Ref` as a class constructor parameter.

Reply:
ah thanks for the reminder, fixed :+1: